### PR TITLE
Improving testing infrastructure

### PR DESCRIPTION
**Pull request by nhibberd**
Just for my comparison
- [x] S3Repository
- [x] ScalaCheckManagedProperties
- [x] *Temporary
- [x] *TemporarySpec
- [x] *Temporary Arbitraries
- [x] cleanup 
- [x] dependencies
- [x] change mr.scala to RIO

### CODE REVIEW

File: ivory-core/src/test/scala/com/ambiata/ivory/core/IvoryConfigurationTemporary.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.ambiata.com.amazonaws.services.s3.AmazonS3Client
+import com.ambiata.mundane.control._
+import com.ambiata.mundane.io._
+import com.ambiata.poacher.hdfs._
+import com.ambiata.saws.core._
+import org.apache.hadoop.conf.Configuration
+import com.nicta.scoobi.Scoobi._
+import scalaz._, Scalaz._, effect.IO
+
+case class IvoryConfigurationTemporary(dir: String, client: AmazonS3Client, configuration: Configuration) { // TODO remove confs
```
Comment (charleso) [inline ivory-core/src/test/scala/com/ambiata/ivory/core/IvoryConfigurationTemporary.scala]: Is this TODO here because `configuration` isn't used?

**Comment by nhibberd** [comment]
If anyone has time :eyes: @domdere @charleso @raronson @markhibberd

File: ivory-storage/src/test/scala/com/ambiata/ivory/storage/repository/TemporaryRepositoriesT.scala [hunk 2]
```diff
@@ -8,6 +7,6 @@ import com.ambiata.notion.core.TemporaryType
 object TemporaryRepositoriesT {
 
   def withRepositoryT[A](temporaryType: TemporaryType)(f: RepositoryTIO[A]): RIO[A] = {
-    TemporaryRepositories.withRepository(temporaryType)(repo => f.toIvoryT(repo).run(IvoryRead.create))
+    ??? //TemporaryRepositories.withRepository(temporaryType)(repo => f.toIvoryT(repo).run(IvoryRead.create))
```
Comment (markhibberd) [inline ivory-storage/src/test/scala/com/ambiata/ivory/storage/repository/TemporaryRepositoriesT.scala]: :bomb:

File: ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/mr.scala [hunk 3]
```diff
@@ -67,7 +67,7 @@ object IngestJob {
     Committer.commit(ctx, {
       case "errors"  => errors
       case "factset" => target
-    }, true).run(conf).run.unsafePerformIO()
+    }, true).run(conf).unsafePerformIO
```
Comment (markhibberd) [inline ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/mr.scala]: I think this is throwing away errors?

File: ivory-storage/src/test/scala/com/ambiata/ivory/storage/fact/NamespacesSpec.scala [hunk 4]
```diff
@@ -31,43 +34,41 @@ class NamespacesSpec extends Specification with ScalaCheck with ScalaCheckManage
...
-  def e1 = prepare { factsetPath =>
-    namespaceSizes(factsetPath).run(new Configuration).map(_.toMap)
-  } must beOkValue(Map(Name("ns1") -> 4.bytes, Name("ns2") -> 4.bytes))
-
-  def e2 = prepare { factsetPath =>
-    namespaceSizesSingle(new Path(factsetPath, "ns1"), Name("namespace")).run(new Configuration)
-  } must beOkValue((Name("namespace"), 4.bytes))
-
-  def e3 = managed { temp: TemporaryDirPath => (nsInc: Set[Name], nsExc: Set[Name], fsInc: FactsetIds, fsExc: FactsetIds) =>
-    !nsInc.exists(nsExc.contains) ==> {
```
Comment (markhibberd) [inline ivory-storage/src/test/scala/com/ambiata/ivory/storage/fact/NamespacesSpec.scala]: Where did this implication go?

**Comment by markhibberd** [comment]
:curry:

File: ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/mr.scala (hunk 3 above)
Comment (charleso) [inline ivory-operation/src/main/scala/com/ambiata/ivory/operation/ingestion/mr.scala]: > I think this is throwing away errors?

https://github.com/ambiata/ivory/pull/555

Feel free to cherry pick if you want Nick

File: ivory-core/src/test/scala/com/ambiata/ivory/core/IvoryLocationTemporary.scala [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+object IvoryLocationTemporary {
+  implicit def IvoryLocationTemporaryArbitrary: Arbitrary[IvoryLocationTemporary] =
+    Arbitrary(arbitrary[T].map(IvoryLocationTemporary(_, s"${java.util.UUID.randomUUID().toString}",
+      IvoryConfiguration.fromConfiguration(new Configuration))))
+
+  /** Deprecated callbacks. Use `IvoryLocationTemporary.file` or `IvoryLocationTemporary.directory` */
+  def withIvoryLocationDir[A](t: T)(f: IvoryLocation => RIO[A]): RIO[A] = for {
+    c <- IvoryConfigurationTemporary.random.conf
+    p = s"${java.util.UUID.randomUUID().toString}"
```
Comment (charleso) [inline ivory-core/src/test/scala/com/ambiata/ivory/core/IvoryLocationTemporary.scala]: Any reason for the string interpolation?

File: ivory-core/src/test/scala/com/ambiata/ivory/core/IvoryLocationTemporary.scala [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+case class IvoryLocationTemporary(t: T, path: String, conf: IvoryConfiguration) {
+  override def toString: String =
+    s"IvoryLocationTemporary($t, $path)"
+
+  def file: RIO[IvoryLocation] =
+    directory
+
+  def directory: RIO[IvoryLocation] = for {
+    l <- LocationTemporary(t, path, conf.s3Client, conf.configuration).location
+    r = locationToIvoryLocation(l)
```
Comment (charleso) [inline ivory-core/src/test/scala/com/ambiata/ivory/core/IvoryLocationTemporary.scala]: Would probably just use map here /supernit

**Comment by charleso** [comment]
:gem:

**Comment by nhibberd** [comment]
rebased :yellow_heart: :on: :libra: :oncoming_automobile:

**Comment by nhibberd** [comment]
- [ ] DictionaryImporterSpec

**Comment by nhibberd** [comment]
rebased, just waiting for builds

**Pull request merged**